src/main/java/uk/ac/cam/cl/dtg/isaac/quiz/IsaacNumericValidator.java

```diff
@@ -251,10 +251,11 @@ private QuantityValidationResponse validateWithUnits(final IsaacNumericQuestion
 
                 // What sort of match do we have:
                 if (numericValuesMatched && unitsFromUser.equals(unitsFromChoice)) {
-                    // Exact match: nothing else can do better.
+                    // Exact match: nothing else can do better, but previous match may tell us if units are also correct:
+                    Boolean unitsCorrect = (null != bestResponse && bestResponse.getCorrectUnits()) || quantityFromQuestion.isCorrect();
                     bestResponse = new QuantityValidationResponse(isaacNumericQuestion.getId(), answerFromUser,
                             quantityFromQuestion.isCorrect(), (Content) quantityFromQuestion.getExplanation(),
-                            quantityFromQuestion.isCorrect(), quantityFromQuestion.isCorrect(), new Date());
+                            quantityFromQuestion.isCorrect(), unitsCorrect, new Date());
```

Comment:
This seems to make more sense than the `quantityFromQuestion...` that was here before